app/src/main/java/com/nextcloud/talk/activities/CallActivity.kt

```diff
@@ -663,7 +675,38 @@ class CallActivity : CallBaseActivity() {
                 ).show()
             }
         }
-        binding!!.hangupButton.setOnClickListener { hangup(true) }
+
+        if (isGroupConversation) {
+            binding!!.hangupButton.setOnClickListener {
+                hangup(true, null)
+            }
+        }
+
+        if (isOneToOneConversation && isModerator) {
+            binding!!.hangupButton.setOnClickListener {
+                showPopupMenu()
+            }
+        }
+
+        if (isOneToOneConversation) {
+            binding!!.hangupButton.setOnLongClickListener {
+                hangup(true, null)
+                true
+            }
+        }
+
+        if (isModerator && isGroupConversation) {
+            binding!!.hangupButton.setOnLongClickListener {
+                showPopupMenu()
+                true
+            }
+        }
+
+        binding!!.popupMenu.setOnClickListener {
+            hangup(true, true)
```

Comment:
Actually, also for the popup button there has to be a check if it is a one-to-one conversation or not, so the behavior of the button is different depending on the type.